### PR TITLE
Add toast component and show when user updates their language preferences

### LINKED ISSUE #130: Add some sort of toast notifications and update ResourceLanguagePreferenceSwitcher to use it for errors

### LINKED ISSUE #135: When a user changes the language in the dropdown, also change it in their user preferences

### PR DESCRIPTION
**Pull request by TammyTee**
This resolves #130 and resolves #135. I see that there was discussion around using cookies to save guest preferences. Wondering if this is already being done and is what's happening in `App\Preferences`?

### CODE REVIEW

File: resources/js/components/LanguageSwitcher.vue [hunk 1]
```diff
@@ -30,16 +34,38 @@
...
         },
 
         methods: {
+            choose(value) {
+                let that = this;
+
+                axios.patch(route('user.preferences.update', { 'locale': 'en' }), {
+                    'locale': value,
+                })
+                .then(function () {
```
Comment (mattstauffer) [inline resources/js/components/LanguageSwitcher.vue]: Can we get around hoisting `this` by using a fat arrow function instead?

```javascript
               .then(() => {
                    let segments = this.domLocation.pathname.split('/');
                    segments[1] = value;
                    window.location = `${this.domLocation.origin}${segments.join('/')}`;
                })
```

File: resources/js/components/Toast.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<template>
+    <notifications position="bottom right">
+        <template slot="body" slot-scope="props">
+            <div class="bg-blue-500 text-white text-xs font-medium py-2 px-3 m-3 mt-0 rounded-sm">
+                <a>{{props.item.title}}</a>
```
Comment (mattstauffer) [inline resources/js/components/Toast.vue]:
```suggestion
                <a>{{ props.item.title }}</a>
```

File: app/Providers/ViewServiceProvider.php [hunk 3]
```diff
@@ -13,6 +13,7 @@ public function boot()
         View::composer('partials.language-switcher', function ($view) {
             $view->with([
                 'localeSlugs' => Localization::slugs(),
```
Comment (mattstauffer) [inline app/Providers/ViewServiceProvider.php]: Do we need `localeSlugs` anymore?

File: app/Http/Controllers/PreferenceController.php [hunk 4]
```diff
@@ -29,6 +29,8 @@ public function update(Request $request)
             auth()->user()->save();
         }
 
+        session()->flash('toast', 'Your preferences were succesfully updated');
```
Comment (mattstauffer) [inline app/Http/Controllers/PreferenceController.php]:
```suggestion
        session()->flash('toast', 'Your preferences were successfully updated');
```

**Comment by TammyTee** [comment]
This should be good to go

File: resources/views/partials/language-switcher.blade.php [hunk 5]
```diff
@@ -1,7 +1,5 @@
-<language-switcher language="{{ $language }}" class="pb-4 md:pb-0">
-    @foreach ($localeSlugs as $locale)
-        <a href="{{ switch_locale_link($locale) }}"
-           class="block px-4 py-2 text-blue-700 hover:bg-blue-700 hover:text-white"
-           style="text-decoration: none">{{ Localization::languageForLocale($locale) }}</a>
-    @endforeach
+<language-switcher
+    language="{{ $language }}"
+    :other-languages="{{ json_encode($locales) }}"
```
Comment (mattstauffer) [inline resources/views/partials/language-switcher.blade.php]: Sorry, one last note! Naming this prop "other languages" makes me think you've excluded this chosen language, but this is really *all* languages. Can you rename it here and in the component to just "languages" and then you can merge this PR? Thanks!

**Pull request merged**